src/helper/rsa.rs

```diff
@@ -0,0 +1,177 @@
+use std::{fmt, marker::PhantomData};
+
+use chrono::{DateTime, Utc};
+use digest::{typenum::Unsigned, OutputSizeUser};
+use rsa::{
+    pkcs1v15::{Signature, VerifyingKey},
+    RsaPublicKey,
+};
+use sha2::Digest;
+use signature::{hazmat::PrehashSigner, Keypair, SignatureEncoding};
+
+use super::{PgpHash, PgpPublicKey};
+use crate::{
+    crypto::{hash::HashAlgorithm, public_key::PublicKeyAlgorithm},
+    errors::{bail, Result},
+    packet::{PubKeyInner, PublicKey},
+    types::{
+        Fingerprint, KeyDetails, KeyId, KeyVersion, Mpi, Password, PublicKeyTrait, PublicParams,
+        RsaPublicParams, SecretKeyTrait, SignatureBytes,
+    },
+};
+
+impl PgpPublicKey for RsaPublicKey {
+    const PGP_ALGORITHM: PublicKeyAlgorithm = PublicKeyAlgorithm::RSA;
+
+    fn pgp_parameters(&self) -> PublicParams {
+        PublicParams::RSA(RsaPublicParams { key: self.clone() })
+    }
+}
+
+/// [`signature::Signer`] backed signer for PGP.
+pub struct RsaSigner<T, D> {
+    inner: T,
+    public_key: PublicKey,
+    _digest: PhantomData<D>,
+}
+
+impl<D, T> fmt::Debug for RsaSigner<T, D> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("RsaSigner").finish()
```

Comment:
I wonder if it makes sense to constrain T and D to implement Debug and then derive Debug.

Alternatively make the debug at least print the public key member. 🤔

Reply:
No we can't because `trait SecretKeyTrait: KeyDetails + std::fmt::Debug {}`, but we can print the publickey